framework/src/meshgenerators/StackGenerator.C

```diff
@@ -108,6 +116,22 @@ StackGenerator::generate()
     second = mesh->get_boundary_info().get_id_by_name(getParam<BoundaryName>("back_boundary"));
   }
 
+  // Check if the user provided boundary ids instead of names
+  if (dim == 3)
+  {
+    if (isParamValid("front_boundary_id"))
+      first = getParam<boundary_id_type>("front_boundary_id");
+    if (isParamValid("back_boundary_id"))
+      second = getParam<boundary_id_type>("back_boundary_id");
+  }
+  if (dim == 2)
```

Comment:
shouldn't this apply to both dim == 2 and 3?

Reply:
When you have a 2D mesh, the boundary names are top, bottom, right and left. Back and front appear only in 3D meshes (at least I think that's what happens).

Reply:
That's correct, but the logic here won't go into the "top/bottom" case when you have a 3D mesh because you have (dim == 2) as your branch condition. Is that what you want?

Reply:
Yes, we don't need top/bottom boundaries in the 3D case